src/sentry/snuba/discover.py

```diff
@@ -978,15 +979,19 @@ def measurements_histogram_query(
         If left unspecified, it is queried using `user_query` and `params`.
     :param float max_value: The maximum value allowed to be in the histogram.
         If left unspecified, it is queried using `user_query` and `params`.
+    :param str data_filter: Indicate the filter strategy to be applied to the data.
     """
 
+    if data_filter not in ("all", "upper_inner_fence", "upper_outer_fence", "p90", "p95", "p99"):
+        data_filter = "upper_inner_fence"
```

Comment:
You could also raise an exception here as the request is for invalid data. I think this smoothing over makes sense at the endpoint or ui layers. Not as much here.